packages/main/src/themes/Switch.css

```diff
@@ -372,11 +373,31 @@
 }
 
 [dir="rtl"].ui5-switch-root .ui5-switch-text--on {
-	right: var(--_ui5_switch_text_active_left);
+	right: var(--_ui5_switch_text_active_left_alternate);
 	left: var(--_ui5_switch_text_active_right);
 }
 
 [dir="rtl"].ui5-switch-root .ui5-switch-text--off {
 	right: var(--_ui5_switch_text_inactive_left);
 	left: var(--_ui5_switch_text_inactive_right);
 }
+
+[dir="rtl"].ui5-switch-root:not(.ui5-switch--checked).ui5-switch--semantic .ui5-switch-text--off,
```

Comment:
I hoped to see solution based on CSS Logical props as it's recommended for a long time now.
Same way  replace
`[dir="rtl"] {
	margin-left: 1rem
}`


with
`margin-inline`

we can replace `left, right` with `inset-inline-start, inset-inline-end`

The Switch is one of the few components to still rely on [dir="rtl"], it's a great opurtinity to get rid of it and remove a lot of CSS.

Reply:
You're right, I am sorry.. I will be more careful and make sure to use the Logical Props always from now on, to secure seemless rtl behavior for our controls as recommended.